src/pages/index.js

```diff
@@ -41,48 +51,110 @@ const enableValidation = (config) => {
 };
 enableValidation(config);
 
-const createCard = (cardObject) => {
-  const card = new Card(cardObject, "#card__template", () => {
-    imagePopup.open(cardObject);
+const handleImageClick = (data) => {
+  imagePopup.open(data);
+};
+
+const handleDelete = (card) => {
+  confirmationPopup.open();
+  confirmationPopup.setSubmitAction(() => {
+    api
+      .deleteCard(card)
+      .then(() => {})
+      .catch((err) => console.error(err));
+    confirmationPopup.close();
+    card.handleTrashBtn();
```

Comment:
You need to **delete** cards only in block `then` after a **successful** response from the server so that if there is an error the card would stay in the `DOM`.